src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -243,19 +245,26 @@ class TezosNodeOperator(val node: TezosRPCInterface, batchConf: BatchFetchConfig
     }
 
     //from the same json, converts to a list of operations and the involved account ids
-    val jsonToOperationsAndAccounts: ((BlockHash, String)) => (BlockHash, List[OperationGroup], List[AccountId]) = {
+    val jsonToOperationsAndAccounts: ((BlockHash, String)) => JsonDecoded[(BlockHash, List[TezosOperations.Group], List[AccountId])] = {
       case (hash, json) =>
-        (hash, jsonToOperationGroups(json), jsonToAccountInvolved(json))
+        jsonToOperationGroups(json).map( groups => (hash, groups, jsonToAccountInvolved(json)))
     }
 
     val isGenesis = (metadata: BlockMetadata) => metadata.header.level == 0
 
+    def decodeOperations(in: List[(BlockHash, String)]): Future[List[(BlockHash, List[TezosOperations.Group], List[AccountId])]] =
```

Comment:
Could we please document this function?

Reply:
I need to add documentation in a few places still, got it!

Reply:
done